src/components/Canvas/BaseLayer.tsx

```diff
@@ -86,5 +89,21 @@ export abstract class BaseLayer {
     this.dpr = devicePixelRatio ? devicePixelRatio : this.dpr;
   }
 
+  setTopRowIndex(topRowIndex: number) {
+    this.topRowIndex = topRowIndex;
+  }
+
+  setLeftColumnIndex(leftColumnIndex: number) {
+    this.leftColumnIndex = leftColumnIndex;
+  }
```

Comment:
This indeed cleaned up the `setPanZoom` logic that is great but leaves me one concern that is if it okay to decouple with dataLayer's rowIndex & columnIndex. Looks like a dependency inversion and wouldn't it be hard to track which index is the real index. Please comment back if my comment is wrong

Reply:
And also with the relationship with rowKeyOrderMap & columnKeyOrderMap, would it be okay to have a setter representing overlapped origin but actual values are differs?

Reply:
Thankyou for your feedback. You seem to have a point. It would be great if we could have a common global variable that all layers can refer to (maybe Singleton pattern..?) to prevent any mismatches between dependencies. But I have not found the best option to handle this.

Currently, if you see the code, `setTopRowIndex` and `setLeftColumnIndex` is only called in Editor's `setLeftTopIndices`, so I believe there will no unpredictable changes in the topRowIndex and leftColumnIndex in each layer.

https://github.com/hunkim98/dotting/blob/e242fa53b72936210d3cd65f079b2bc68ace251c/src/components/Canvas/Editor.tsx#L806-L815

I am not usre how to handle this. Do you have any ideas on how to better this?

Reply:
Discussed about this offline, decided to make it possible to track data inside the dataLayer in a top-down context. Would you like to merge this PR and address it in the next PR?

Reply:
Yes I will do so! @Lee-Si-Yoon Thankyou